File: java/pe003.java
//file: pe003.java

import java.math.BigInteger;

public class pe003 {
  public static void main( String[] args ) {
    BigInteger big_number = new BigInteger("600851475143");
    int max_prime = 0;
    int counter = 2;
    while (big_number.intValue() != 1) {
      if (big_number.mod(BigInteger.valueOf(counter)).intValue() == 0) {
        max_prime = counter;
        big_number = big_number.divide(BigInteger.valueOf(counter));
      } else {
        counter++;
      }
    }
    System.out.println(max_prime);
  }
}

File: typescript/pe003.ts
function pe003(num03:number): number {
  let max_prime:number = 0
  let i:number = 3

  while (num03 != 1)
  {
    if (num03 % i)
      i+=2;
    else
    {
      max_prime = i;
      num03 = num03/i;
    }
  }
  return max_prime
}

let num03:number = 600851475143
console.log(pe003(num03));
